consensus/common/src/main/java/org/hyperledger/besu/consensus/common/bft/BftExtraDataEncoder.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.hyperledger.besu.consensus.common.bft;
+
+import org.hyperledger.besu.ethereum.core.BlockHeader;
+
+import org.apache.tuweni.bytes.Bytes;
+
+public interface BftExtraDataEncoder {
+
+  int EXTRA_VANITY_LENGTH = 32;
+
+  Bytes encode(BftExtraData extraData);
+
+  Bytes encodeWithoutCommitSeals(BftExtraData bftExtraData);
+
+  Bytes encodeWithoutCommitSealsAndRoundNumber(BftExtraData bftExtraData);
+
+  BftExtraData decode(final BlockHeader blockHeader);
```

Comment:
nit: no need for final modifier in an interface.

Reply:
done